soracom/orbit.c

```diff
@@ -146,15 +146,55 @@ int32_t soracom_get_userdata_as_string(const char** buf, size_t* siz) {
     return ERR_OK;
 }
 
+/**
+ * Get the original request as a null terminated string.
+ *
+ * @param buf [out] A pointer to the allocated memory.
+ * @param siz [out] The length of the string.
+ *
+ * @note The pointer received in buf should be released by calling soracom_release_buffer().
+ *
+ * @return ERR_OK: successfully retrieved the userdata.
+ *         ERR_INVALID_ARG: buf or siz is null.
+ */
+
+int32_t soracom_get_original_request_as_string(const char** buf, size_t* siz) {
+    if ((buf == NULL) || (siz == NULL)) {
+        return ERR_INVALID_ARG;
+    }
+
+    *siz = orbit_get_original_request_len();
+    *buf = (const char*) malloc(*siz + 1);
+    memset((void*)(*buf), 0, *siz + 1);
+    orbit_get_original_request(*buf, *siz);
+    return ERR_OK;
+}
+
+
+/**
+ * Release the buffer memory.
+ *
+ * @param buf [in] The pointer to the allocated
+ *
+ * @note The pointer buf is allocated by 
+ * - soracom_get_userdata_as_string().
+ * - soracom_get_original_request_as_string().
+ */
+void soracom_release_buffer(const char* buf) {
+    free((void*)(buf));
+}
+
 /**
  * Release the userdata.
  *
  * @param buf [in] The pointer to the allocated userdata.
  *
  * @note The pointer buf is allocated by soracom_get_userdata_as_string().
+ * @deprecated Use soracom_release_buffer instead
+ * 
```

Comment:
is there any better way to set a function as deprecated?

tried: https://en.cppreference.com/w/c/language/attributes/deprecated

but there is no warning whatsoever on build. on vscode there is no strikethrough

Reply:
Which version of emcc did you use? It seems that the `[[deprecated]]` attribute is supported after C23 and emcc ver >= 3.0.0 has support for C23 standard.

Reply:
my environment

```
sen@WorkMBP code % emcc --check
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.73-git
shared:INFO: (Emscripten: Running sanity checks)
sen@WorkMBP code % clang --version
Apple clang version 16.0.0 (clang-1600.0.26.4)
Target: arm64-apple-darwin24.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```